### PR TITLE
chore: Link contributing doc partial to CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by gerhard**
This makes it easier for people that are familiar with GitHub and expect `CONTRIBUTING.md` to  exist. Here is a recent example when a new contributor to our repository is likely to have found this useful: https://github.com/dagger/dagger/pull/4094#issuecomment-1341048416

While this was available via https://docs.dagger.io, it is not easy to find: https://docs.dagger.io/sdk/cue/749394/contributing

Also, GitHub is nudging us into adding this via https://github.com/dagger/dagger/community. This change looks to me as if it's going to make everyone happy 🙌

Screenshot 2022-12-07 at 20 13 30

---

### Questions
- [x] Will Docusaurus work if that partial is a symlink? [Yes, according to the preview build.](https://6390f2c50ea53400092cdb14--devel-docs-dagger-io.netlify.app/sdk/cue/749394/contributing/#github-workflow)
- [ ] Should I cleanup `CONTRIBUTING.md` part of this PR? I think so, but don't wan to go too far with this change if we don't like it. I would remove the SOPS FAQ at the end of the doc.

### CODE REVIEW

**Review by vikram-dagger — APPROVED** [review]
This lgtm, but perhaps it could be made better/more visible by also adding a standalone page in the current docs sidebar (same level as "FAQ") named "Contribution Guidelines" or similar and embedding the partial. This would be similar to the page/approach used in the 0.2 docs and IMO would make it significantly easier to find.

**Comment by gerhard** [comment]
> perhaps it could be made better/more visible by also adding a standalone page in the current docs sidebar (same level as "FAQ") named "Contribution Guidelines" or similar and embedding the partial.

That is a good idea! Is this what you had in mind @vikram-dagger? https://63910468d2415e00089b3d7b--devel-docs-dagger-io.netlify.app/

Screenshot 2022-12-07 at 21 25 38

What are your thoughts on this change? https://github.com/dagger/dagger/pull/4127/commits/bbf19b86682b64ba7cd7c854b124749a10d82a2e

File: docs/current/sdk/nodejs/reference/modules.md [hunk 1]
```diff
@@ -11,4 +11,5 @@ displayed_sidebar: "current"
 ## Modules
 
 - [api/client.gen](modules/api_client_gen.md)
+- [common/errors](modules/common_errors.md)
```
Comment (vikram-dagger) [inline docs/current/sdk/nodejs/reference/modules.md:14]: This is the issue I mentioned earlier - this file is auto-generated when the website is tested locally and pollutes the commit

**Review by vikram-dagger — APPROVED** [review]

File: docs/current/sdk/nodejs/reference/modules.md (hunk 1 above)
Comment (gerhard) [reply docs/current/sdk/nodejs/reference/modules.md]: I remember.

I am going to drop this change & run `git update-index --skip-worktree docs/current/sdk/nodejs/reference/` locally:

image

I am think that we should gitignore that entire path, and stop tracking generated files in git. Is there a reason why we are not doing that already?

Comment (gerhard) [reply docs/current/sdk/nodejs/reference/modules.md]: image

Comment (vikram-dagger) [reply docs/current/sdk/nodejs/reference/modules.md]: > I am think that we should gitignore that entire path, and stop tracking generated files in git. Is there a reason why we are not doing that already?

The sidebar includes a hard link to reference/modules.md so it was not possible to gitignore it at the very beginning, as that would have excluded it from the repository and Docusaurus would have thrown a broken link error. However, now that the file is part of the repository, we could indeed add it to .gitignore so future changes to it are not tracked.

**Pull request merged**